### PR TITLE
treewide: fix redirected and broken URLs

### PR DESCRIPTION
**Pull request by anthonyroussel**
## Description of changes

Fix 3xx and 4xx URLs in packages, documentation, modules
(Sorry for the ping, it is a treewide pull-request)

Using the script in maintainers/scripts/update-redirected-urls.sh

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/yate/default.nix [hunk 1]
```diff
@@ -33,7 +33,7 @@ stdenv.mkDerivation rec {
 
   meta = {
     description = "Yet another telephony engine";
-    homepage = "http://yate.ro/";
+    homepage = "https://yate.ro/";
```
Comment (anthonyroussel) [inline pkgs/applications/misc/yate/default.nix:36]: Also fix the broken download yate.null.ro and switch src to Github https://github.com/yatevoip/yate

File: pkgs/development/libraries/ctpp2/default.nix [hunk 2]
```diff
@@ -5,7 +5,7 @@ stdenv.mkDerivation rec {
   version = "2.8.3";
```
Comment (anthonyroussel) [inline pkgs/development/libraries/ctpp2/default.nix:5]: could be upgraded to v2.9.0 in another PR

File: pkgs/development/php-packages/psalm/composer.lock [hunk 3]
```diff
@@ -675,7 +675,7 @@
                 {
                     "name": "Christian Weiske",
                     "email": "cweiske@cweiske.de",
-                    "homepage": "http://github.com/cweiske/jsonmapper/",
+                    "homepage": "https://github.com/cweiske/jsonmapper/",
```
Comment (drupol) [inline pkgs/development/php-packages/psalm/composer.lock:678]: Can you please revert this change? This file is not meant to be manually modified and this shouldn't be touched. There's a lot of chance that touching this will have an impact since the checksum of the file would be different.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/development/php-packages/psalm/composer.lock (hunk 3 above)
Comment (anthonyroussel) [reply pkgs/development/php-packages/psalm/composer.lock]: Oops, thanks for the review

**Review by drupol — APPROVED** [review]
Thanks, LGTM now :)

**Comment by alyssais** [comment]
node-packages.nix is another generated file that probably shouldn't be directly modified.

**Review by alyssais — APPROVED** [review]

**Review by pbsds — APPROVED** [review]

File: maintainers/scripts/haskell/upload-nixos-package-list-to-hackage.sh [hunk 4]
```diff
@@ -39,5 +39,5 @@ fi
 package_list="$(nix-build -A haskell.package-list)/nixos-hackage-packages.csv"
 username=$(grep "^username:" "$CABAL_DIR/config" | sed "s/^username: //")
 password_command=$(grep "^password-command:" "$CABAL_DIR/config" | sed "s/^password-command: //")
-curl -u "$username:$($password_command | head -n1)" --digest -H "Content-type: text/csv" -T "$package_list" http://hackage.haskell.org/distro/NixOS/packages.csv
+curl -u "$username:$($password_command | head -n1)" --digest -H "Content-type: text/csv" -T "$package_list" https://hackage.haskell.org/distro/NixOS/packages.csv
```
Comment (sternenseemann) [inline maintainers/scripts/haskell/upload-nixos-package-list-to-hackage.sh:42]: Thanks!

File: pkgs/applications/editors/textadept/deps.nix [hunk 5]
```diff
@@ -36,7 +36,7 @@
   };
   # cdk
   "t20200923.tar.gz" = {
-    url = "http://github.com/ThomasDickey/cdk-snapshots/archive/refs/tags/t20200923.tar.gz";
```
Comment (sternenseemann) [inline pkgs/applications/editors/textadept/deps.nix:39]: This file is likely generated,so this will be reverted.

Comment (anthonyroussel) [reply pkgs/applications/editors/textadept/deps.nix]: Thanks for the review. I reverted this change.

**Comment by anthonyroussel** [comment]
Solved conflicts

**Comment by anthonyroussel** [comment]
Solved conflicts against master

**Comment by anthonyroussel** [comment]
Hello! Would anyone be available to do a final review and merge this PR, please? :)

File: nixos/modules/services/misc/mediatomb.nix [hunk 6]
```diff
@@ -186,7 +186,7 @@ let
   defaultFirewallRules = {
     # udp 1900 port needs to be opened for SSDP (not configurable within
     # mediatomb/gerbera) cf.
-    # http://docs.gerbera.io/en/latest/run.html?highlight=udp%20port#network-setup
+    # https://docs.gerbera.io/en/latest/run.html#network-setup
```
Comment (pbsds) [inline nixos/modules/services/misc/mediatomb.nix:189]: This one strips highlighting

Comment (anthonyroussel) [reply nixos/modules/services/misc/mediatomb.nix]: Fixed, thanks for the review
